### PR TITLE
add permissions documentation

### PR DESCRIPTION
**Pull request by AurelienGasser**

### CODE REVIEW

**Comment by AurelienGasser** [comment]
@natct10 what else is needed for the build to pass? I ran `make docs` and pushed the changes, but it doesn't seem to help.

**Comment by natct10** [comment]
> @natct10 what else is needed for the build to pass? I ran `make docs` and pushed the changes, but it doesn't seem to help.

Hi, this looks great, thank you @AurelienGasser!  I don't know why the github action is not triggered :/

**Comment by natct10** [comment]
> > @natct10 what else is needed for the build to pass? I ran `make docs` and pushed the changes, but it doesn't seem to help.
> 
> Hi, this looks great, thank you @AurelienGasser! I don't know why the github action is not triggered :/

Could you please try again `make docs` and `git add .` (+ push)? In the meantime, I am adding you you to the repo so you don't have to use a fork

File: src/permissions.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Permissions
+
+Assets are protected using a permission system.
+
+Permissions are either specified  by the user at asset registration time, or they are automatically inherited from dependent assets (see “Permission inheritance”).
```
Comment (mblottiere) [inline src/permissions.md:5]:
```suggestion
Permissions are either specified by the user at asset registration time, or they are automatically inherited from dependent assets (see “Permission inheritance”).
```

File: src/permissions.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+Permissions are structured into two properties:
+
+- *Public* (bool): if True, the asset is downloadable/processable *by all nodes*  in the channel.
+- *Authorized ids* (list of strings): a *list of nodes* allowed to process/download the asset.
+
+Using this structure, there are 3 possible use-cases:
+
+- the asset is downloadable or processable by **all nodes** in the network => `Public = True`
+- the asset is downloadable or processable by **a set of nodes** in the network, listed by their identity -> `Public = True, Authorized_Ids = [NodeA, NodeB]`
```
Comment (mblottiere) [inline src/permissions.md:17]: The example does not match the description:
```suggestion
- the asset is downloadable or processable by **a set of nodes** in the network, listed by their identity -> `Public = False, Authorized_Ids = [NodeA, NodeB]`
```

File: src/permissions.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+- the asset is only available for its owner => `Public  = False, Authorized_Ids = [MyNode]`
+
+
+## Process vs Download
+
+There are two types of permissions:
+  - **download**: possibility to download the asset, for users of a node
+  - **process**: possibility to use the asset in a node on the platform without downloading it
+
+Note that if a node has no permissions to process an asset, it won't have permissions to download it.  Conversely, permissions to download imply permissions to process.
```
Comment (mblottiere) [inline src/permissions.md:27]:
```suggestion
Note that if a node has no permissions to process an asset, it won't have permissions to download it.
Conversely, permissions to download imply permissions to process.
```

File: src/permissions.md (hunk 2 above)
Comment (AurelienGasser) [reply src/permissions.md:17]: Good catch!

**Comment by AurelienGasser** [comment]
Closed in favor of https://github.com/SubstraFoundation/substra-documentation/pull/161

**Review by RomainGoussault — COMMENTED** [review]
Some minor comments.
Very nice and easy to read @AurelienGasser !

File: src/permissions.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+## Permissions by asset type
+
+Unless stated otherwise, items in this list refer to *Process* permission.
+
+- **Dataset**: Permission to use this dataset for training
+- **Algo**: Permission to use this algo for training
+- **Traintuple, Aggregatetuple**: Permission to process/download the out-model
+- **Composite Traintuple**: Permission to process/download the trunk out-model. For the head out-model, see “Permission inheritance”.
+- **Model** (out-models)
+  - *Process*: Permission to use this model as an *inModel*
```
[the remark this answers is not in the document]
Comment (AurelienGasser) [reply src/permissions.md:42]: I'd say 
- in-model(s) / out-model(s) in plain english
- `in_model` / `out_model` when referring to code

What do you think?

File: src/permissions.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
+# Permissions
+
+Assets are protected using a permission system.
```
[the remark this answers is not in the document]
Comment (AurelienGasser) [reply src/permissions.md:3]: Yes, I agree it's better although I'd remove "in Substra"

File: src/permissions.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+## Updating permissions
+
+**Permissions are immutable**: it is not possible to update permissions.
+
+## Permissions by asset type
+
+Unless stated otherwise, items in this list refer to *Process* permission.
+
+- **Dataset**: Permission to use this dataset for training
+- **Algo**: Permission to use this algo for training
```
[the remark this answers is not in the document]
Comment (AurelienGasser) [reply src/permissions.md:38]: As of today: yes

File: src/permissions.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+## Updating permissions
+
+**Permissions are immutable**: it is not possible to update permissions.
+
+## Permissions by asset type
+
+Unless stated otherwise, items in this list refer to *Process* permission.
+
+- **Dataset**: Permission to use this dataset for training
```
[the remark this answers is not in the document]
Comment (AurelienGasser) [reply src/permissions.md:37]: Good point, I'll add it

File: src/permissions.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+- **Algo**: Permission to use this algo for training
+- **Traintuple, Aggregatetuple**: Permission to process/download the out-model
+- **Composite Traintuple**: Permission to process/download the trunk out-model. For the head out-model, see “Permission inheritance”.
+- **Model** (out-models)
+  - *Process*: Permission to use this model as an *inModel*
+  - *Download*: Permission to export this model
```
[the remark this answers is not in the document]
Comment (AurelienGasser) [reply src/permissions.md:43]: Sounds good, maybe replace "laptop/workstation" with "computer"?

File: src/permissions.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (RomainGoussault) [reply src/permissions.md:3]: Yes without Substra 👍

File: src/permissions.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (RomainGoussault) [reply src/permissions.md:42]: I agree!

File: src/permissions.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (RomainGoussault) [reply src/permissions.md:43]: Ah yes! I was looking for a better term, somehow I forgot the word "computer" did exist 🤦.

File: src/permissions.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (RomainGoussault) [reply src/permissions.md:38]: So we can add somethink like: `Permission to use this algo for training, and to download it from a node to the user computer?`
